vendor: bump arrow

asubiotto opened the pull request:
This picks up a fix for power pc arch and a fix for pointer arithmetic that
would cause go 1.14 to complain when running with race enabled.
golang.org/x/xerrors is a new dependency imported by arrow, so it is added as
well.

Release note: None

I'll wait for #49158 to merge

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/49184)
<!-- Reviewable:end -->

yuzefovich reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/49184#-:-M7eSXUxBpR-vmQIpWUg:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @dt)



<!-- Sent from Reviewable.io -->

Commit:
vendor: bump arrow

To bfb07ad1ed6447d74cfcd00682fed70ec01fbdaf.

This picks up a fix for power pc arch and a fix for pointer arithmetic that
would cause go 1.14 to complain when running with race enabled.
golang.org/x/xerrors is a new dependency imported by arrow, so it is added as
well.

Release note: None

asubiotto commented:
bors r=yuzefovich
